craft_application/application.py

```diff
@@ -458,6 +460,9 @@ def run(self) -> int:  # noqa: PLR0912 (too many branches)
             )
             platform = getattr(dispatcher.parsed_args(), "platform", None)
             build_for = getattr(dispatcher.parsed_args(), "build_for", None)
+            provider_name = (
+                "lxd" if getattr(dispatcher.parsed_args(), "use_lxd", None) else None
```

Comment:
Are you wanting to use `command.provider_name(dispatcher.parsed_args())` here?

If not, what's actually calling the command's `provider_name`?

Reply:
Thanks, fixed.